mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -88,36 +88,14 @@ def on_train_start(self, trainer, pl_module):
                     self._log_early_stop_params(callback)
 
             if hasattr(trainer, "optimizers"):
-                for optimizer in trainer.optimizers:
-                    try_mlflow_log(mlflow.log_param, "optimizer_name", type(optimizer).__name__)
-                    optimizer_name = type(optimizer).__name__.lower() + "_optimizer"
-
-                    if hasattr(optimizer, "defaults"):
-                        optim_dict = optimizer.defaults
-
-                        if "lr" in optim_dict:
-                            try_mlflow_log(
-                                mlflow.log_param,
-                                "learning_rate_" + optimizer_name,
-                                optim_dict["lr"],
-                            )
-
-                        if "eps" in optim_dict:
-                            try_mlflow_log(
-                                mlflow.log_param, "epsilon_" + optimizer_name, optim_dict["eps"]
-                            )
-
-                        if "betas" in optim_dict:
-                            try_mlflow_log(
-                                mlflow.log_param, "betas_" + optimizer_name, optim_dict["betas"]
-                            )
-
-                        if "weight_decay" in optim_dict:
-                            try_mlflow_log(
-                                mlflow.log_param,
-                                "weight_decay_" + optimizer_name,
-                                optim_dict["weight_decay"],
-                            )
+                optimizer = trainer.optimizers[0]
+                try_mlflow_log(mlflow.log_param, "optimizer_name", type(optimizer).__name__)
+
+                if hasattr(optimizer, "defaults"):
```

Comment:
@shrinath-suresh Please add some detailed comments for the logic we are putting to only log for the 1st optimizer, so that we can evaluate in future if the behaviour needs to be changed based on user feedback